pkg/bindings/containers/containers.go

```diff
@@ -390,3 +390,15 @@ func ContainerInit(ctx context.Context, nameOrID string) error {
 	}
 	return response.Process(nil)
 }
+
+func ShouldRestart(ctx context.Context, nameOrID string) (bool, error) {
```

Comment:
Why do we need this in the API and Bindings? This should be local-only, `podman container cleanup` should never be invoked remotely

Reply:
In pkg/domain/infra/tunnel/containers.go when we do a non detached container, we Remove the container when it completes. Added this ShouldRestart check to make sure that this does not happen.

Reply:
I don't know if it's worth adding a complete new API endpoint to handle this. Maybe we should just disable removing the container in that case if it detects that the container has this restart policy?

Reply:
Then we end up with the race conditions again.  But for only a small subset of calls.

Reply:
I removed the bindings and just stopped rm the container in run if the restart policy is set to something other then no.

Reply:
Removing the bindings proved to fail testing. So I added them back.